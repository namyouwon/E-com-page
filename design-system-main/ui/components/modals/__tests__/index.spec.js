// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import {
  Taglines,
  ModalSizes,
  Directional,
  Headless,
  Footless,
  HiddenFooter,
  HeadlessAndFootless,
  HeadlessAndFootlessDeprecated,
  Menu,
  DisabledClose,
  DeprecatedClose
} from '../base/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Modal', () => {
  it('renders a default modal', () => matchesMarkup(<defaultComponent />));

  it('renders a modal with a tagline', () => matchesMarkup(<Taglines />));

  it('renders a modal that is small', () =>
    matchesMarkup(<ModalSizes size="small" />));

  it('renders a modal that is medium', () =>
    matchesMarkup(<ModalSizes size="medium" />));

  it('renders a modal that is large', () =>
    matchesMarkup(<ModalSizes size="large" />));

  it('renders a modal that is directional', () =>
    matchesMarkup(<Directional />));

  it('renders a modal that is Headless', () => matchesMarkup(<Headless />));

  it('renders a modal that is Footless', () => matchesMarkup(<Footless />));

  it('renders a modal that has a hidden footer', () =>
    matchesMarkup(<HiddenFooter />));

  it('renders a modal that is Headless and Footless', () => matchesMarkup(<HeadlessAndFootless />));

  it('renders a modal that is Headless and Footless - deprecated', () => matchesMarkup(<HeadlessAndFootlessDeprecated />));

  it('renders a modal that contains a menu', () => matchesMarkup(<Menu />));

  it('renders a modal that has a disabled close button', () => matchesMarkup(<DisabledClose />));

  it('renders a modal that is using deprecated markup for the close button', () => matchesMarkup(<DeprecatedClose />));
});
