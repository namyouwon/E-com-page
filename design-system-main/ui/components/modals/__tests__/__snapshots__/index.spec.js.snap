// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders a default modal 1`] = `<defaultComponent />`;

exports[`Modal renders a modal that contains a menu 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-unique-id-1"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-unique-id-1"
        >
          Modal header
        </h2>
      </div>
      <div
        className="slds-modal__menu"
        id="modal-content-unique-id-2"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer slds-hide"
      />
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that has a disabled close button 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={true}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that has a hidden footer 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-modal__content_has-hidden-footer slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is Footless 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is Headless 1`] = `
Array [
  <section
    aria-label="Meaningful description of the modal content"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__content slds-p-around_medium slds-modal__content_headless"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer"
      >
        <button
          aria-label="Cancel and close"
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is Headless and Footless - deprecated 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header slds-modal__header_empty"
      />
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is Headless and Footless 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__content slds-modal__content_headless slds-modal__content_footless slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is directional 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer slds-modal__footer_directional"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Skip This Step
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save & Next
        </button>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is large 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open slds-modal_large"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer"
      >
        <button
          aria-label="Cancel and close"
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is medium 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open slds-modal_medium"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer"
      >
        <button
          aria-label="Cancel and close"
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is small 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open slds-modal_small"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer"
      >
        <button
          aria-label="Cancel and close"
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal that is using deprecated markup for the close button 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <div
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Cancel and close
          </span>
        </button>
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;

exports[`Modal renders a modal with a tagline 1`] = `
Array [
  <section
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <button
        className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_large"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Cancel and close
        </span>
      </button>
      <div
        className="slds-modal__header"
      >
        <h1
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h1>
        <p
          className="slds-m-top_x-small"
        >
          Here’s a tagline if you need it. It is allowed to extend across mulitple lines, so I’m making up content to show that to you. It is allowed to
           
          <a
            href="#"
            onClick={[Function]}
          >
            contain links or be a link
          </a>
          .
        </p>
      </div>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <div
        className="slds-modal__footer"
      >
        <button
          aria-label="Cancel and close"
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
    role="presentation"
  />,
]
`;
